### PR TITLE
Fix for OpenGL version selection on Windows

### PR DESCRIPTION
**Pull request by EdgarReynaldo**
There are some patches to remove old code, comment out and fix old code, and code to use al_get_new_display_option instead of al_get_suggested_display_option, which was broken. Please see the patch for details. Sorry I'm stupid and made the pull request on my own repository.

### CODE REVIEW

**Comment by fatcerberus** [comment]
> Sorry I'm stupid and made the pull request on my own repository.

That's why I asked about that in the other issue.  All good, it's in the right place now so that's all that matters 😸 

Any possibility you could squash out that merge commit?

**Comment by EdgarReynaldo** [comment]
There I should have removed the extraneous commits

File: src/x/xglx_config.c [hunk 1]
```diff
@@ -532,10 +532,8 @@ bool _al_xglx_config_create_context(ALLEGRO_DISPLAY_XGLX *glx)
          existing_ctx = (*existing_dpy)->context;
    }
 
-   int major = _al_get_suggested_display_option(disp,
-      ALLEGRO_OPENGL_MAJOR_VERSION, 0);
-   int minor = _al_get_suggested_display_option(disp,
-      ALLEGRO_OPENGL_MINOR_VERSION, 0);
+   int major = al_get_new_display_option(ALLEGRO_OPENGL_MAJOR_VERSION , 0);
```
Comment (SiegeLord) [inline src/x/xglx_config.c]: Does this work correctly on X now (assuming it didn't before)?

File: src/display_settings.c [hunk 2]
```diff
@@ -47,17 +47,21 @@ void al_set_new_display_option(int option, int value, int importance)
    extras->settings[option] = value;
 }
 
-
+/**
+  This function uses display->extra_settings, where al_get_new_display_option uses tls->new_display_settings
+  TODO : Remove
```
Comment (SiegeLord) [inline src/display_settings.c]: Why keep it?

File: src/opengl/extensions.c [hunk 3]
```diff
@@ -794,11 +785,24 @@ void _al_ogl_manage_extensions(ALLEGRO_DISPLAY *gl_disp)
    load_extensions(ext_api);
    gl_disp->ogl_extras->extension_api = ext_api;
    
+   
+   /* Print out OpenGL extensions for all versions of opengl less than 3
+    * Does this include opengles 2 and 3?
+    * glGetString(GL_EXTENSIONS) was deprecated in OpenGL 3.0
+    * We should use glGetStringi(GL_EXTENSIONS, i) for OpenGL 3.0+
+    * but it doesn't seem to work until later. Note - it doesn't work for some? Who?
```
Comment (SiegeLord) [inline src/opengl/extensions.c]: I don't understand this last line.

**Comment by EdgarReynaldo** [comment]
This pull request is superceded by https://github.com/liballeg/allegro5/pull/1021
Closing request.

**Pull request closed without merging**